lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -727,6 +727,7 @@ sub module_fails {
             job_id => $self->param('testid'),
             name => $self->param('moduleid'),
         })->first;
+    return $self->reply->not_found unless defined $module;
```

Comment:
I'm with @perlpunk here. I think this would be good to cover in a unit test

Reply:
added test